File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
       int reverce = 0,r;
       boolean isNegative = x < 0;
       if(isNegative){
        x = - x;
       }
        while (x > 0) {
            r = x % 10;
            x = x / 10;
        if (reverce > Integer.MAX_VALUE / 10 || (reverce == Integer.MAX_VALUE / 10 && r > 7)) {
                return 0;
            }
             reverce = reverce * 10 + r;
        }
        if(isNegative){
            reverce = - reverce;
        }
        return reverce;   
    }
}

File: 0007-reverse-integer/0007-reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    let reverse = 0, r = 0;
    let isNegative = false;

    if(x < 0){
        x = - x;
        isNegative = true;
    }
    while (x > 0) {
        reverse = reverse * 10 + (x % 10);
        x = Math.floor(x / 10);
    }
    if(isNegative){
        reverse = - reverse;
    }
    return (reverse >= -(2 ** 31) && reverse <= (2 ** 31 - 1)) ? reverse : 0;
};